app/src/test/java/com/android/sample/feature/camera/scan/HandleBarcodeDetectionTest.kt

```diff
@@ -0,0 +1,105 @@
+package com.android.sample.feature.camera.scan
+
+import com.android.sample.feature.camera.handleBarcodeDetection
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.never
+import org.mockito.Mockito.times
+import org.mockito.kotlin.verify
+
+class HandleBarcodeDetectionTest {
+
+  @Test
+  fun `handleBarcodeDetection does nothing with null barcode`() {
+
+    val recentBarcodes = mutableListOf<Long>()
+    val lastScannedBarcode: Long? = null
+    val onBarcodeDetected: (Long) -> Unit = {}
+
+    handleBarcodeDetection(
+        barcodeValue = null,
+        recentBarcodes = recentBarcodes,
+        lastScannedBarcode = lastScannedBarcode,
+        onBarcodeDetected = onBarcodeDetected)
+
+    // Verify that recentBarcodes remains empty when barcodeValue is null
+    assertTrue(recentBarcodes.isEmpty())
+  }
+
+  @Test
+  fun `handleBarcodeDetection doesn't trigger onBarcodeDetected with different barcode`() {
+
+    val recentBarcodes = mutableListOf<Long>()
+    val lastScannedBarcode: Long? = null
+    val onBarcodeDetected: (Long) -> Unit = mock()
+
+    handleBarcodeDetection(
+        barcodeValue = 1L,
+        recentBarcodes = recentBarcodes,
+        lastScannedBarcode = lastScannedBarcode,
+        onBarcodeDetected = onBarcodeDetected)
+
+    handleBarcodeDetection(
+        barcodeValue = 2L,
+        recentBarcodes = recentBarcodes,
+        lastScannedBarcode = lastScannedBarcode,
+        onBarcodeDetected = onBarcodeDetected)
+
+    handleBarcodeDetection(
+        barcodeValue = 3L,
+        recentBarcodes = recentBarcodes,
+        lastScannedBarcode = lastScannedBarcode,
+        onBarcodeDetected = onBarcodeDetected)
+
+    // Retry with the first barcode
+    handleBarcodeDetection(
+        barcodeValue = 1L,
+        recentBarcodes = recentBarcodes,
+        lastScannedBarcode = lastScannedBarcode,
+        onBarcodeDetected = onBarcodeDetected)
```

Comment:
You can put these values in the `C` file

Reply:
Since this is a test file, I believe handling magic values may not be critical in this context.

Reply:
Yeah my bad, long day, didn't see that it was a test file.